### PR TITLE
Blockfrost metadata and confirmation

### PR DESCRIPTION
**Pull request by jy14898**
Part of #1365

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [ ] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [ ] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

File: src/Internal/QueryM/Kupo.purs [hunk 1]
```diff
@@ -156,19 +165,25 @@ isTxConfirmedAff config (TransactionHash txHash) = runExceptT do
...
-      -- `KupoMetadata` will fail parsing if there is more than one response
-      pure $ unwrapKupoMetadata metadata
+      kupoMetadata <- ExceptT $
+        lmap GetTxMetadataClientError <<< handleAffjaxResponse <$>
+          kupoGetRequest
+            endpoint
+      case unwrapKupoMetadata kupoMetadata of
+        Nothing -> throwError GetTxMetadataMetadataEmptyOrMissingError
+        Just metadata
+          | Map.isEmpty (unwrap metadata) -> throwError
```
Comment (uhbif19) [inline src/Internal/QueryM/Kupo.purs:184]: Why we need to handle empty Map case?

Comment (jy14898) [reply src/Internal/QueryM/Kupo.purs:184]: To make the API the same for blockfrost and kupo.

Blockfrost always gives a response for metadata, regardless of whether it actually had the optional metadata field attached. Kupo on the other hand returns exactly what the transaction contained,

The alternative could be to always make the kupo version return an empty metadata on the case where there was none, but I think if users are requesting metadata from a transaction, they probably expect some metadata.

Comment (jy14898) [reply src/Internal/QueryM/Kupo.purs:184]: Not sure if I explained that well, maybe this makes it clearer:

No metadata:
blockfrost: []
kupo: []

Empty metadata:
blockfrost: []
kupo: [ ... ]

With metadata:
blockfrost: [ ... ]
kupo: [ ... ]

File: src/Internal/Service/Blockfrost.purs [hunk 2]
```diff
@@ -88,14 +88,14 @@ runBlockfrostServiceM backend = flip runReaderT serviceParams
 --------------------------------------------------------------------------------
 
 data BlockfrostEndpoint
-  = GetTransaction TransactionHash
-  | GetTransactionMetadata TransactionHash
+  = Transaction TransactionHash
```
Comment (uhbif19) [inline src/Internal/Service/Blockfrost.purs:91]: Why change from `Get`?

**Review by klntsky — APPROVED** [review]
Looks good

**Review by klntsky — APPROVED** [review]
Looks good to me

**Pull request merged**